pkg/types/balance.go

```diff
@@ -7,11 +7,13 @@ import (
 	"math/big"
 
 	uint128 "github.com/eteu-technologies/golang-uint128"
+	"github.com/shopspring/decimal"
```

Comment:
Not very happy with introducing this dependency without an explanation

Reply:
This dependency library is very reliable. Most of the conversion and calculation operations involving floating-point numbers in our project are solved by using it.

Reply:
But what problem does it solve here?